### PR TITLE
Cookbook for RBAC

### PR DESCRIPTION
**Pull request by dthyresson**
Adds a cookbook for Role-based Access Control and code sample how-to's.

### CODE REVIEW

**Comment by dthyresson** [comment]
Reworked intro text and headings.

Added supporting text to most of the how-to's and also added a new example of extracting roles from a database to set on currentUser.

Still draft.

**Comment by dthyresson** [comment]
Hi all. This RBAC cookbook is ready for review. It has examples for the hasRole([roles]) support coming in v17.

The only thing I may want to add/try/change is that can you prevent rendering of a component or cell using a role?

Is that something people might do ... instead of obfuscating the markup to render nothing?

**Comment by thedavidprice** [comment]
@jtoar this is ready for review. I've assigned it to you. Just let me know otherwise as I'd be happy to help.

Thanks again @dthyresson!

File: docs/authentication.md [hunk 1]
```diff
@@ -652,11 +652,15 @@ You must follow the ["Before you begin"](https://firebase.google.com/docs/auth/w
 
 +++ View Firebase Options
 
-#### Role-based access control (RBAC)
+#### Role-based access control (RBAC) Firebase
```
Comment (thedavidprice) [inline docs/authentication.md:655]:
```suggestion
#### Role-based access control (RBAC) in Firebase
```

File: cookbook/Role-based_Access_Control.md [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+
+When thinking about security, it helps to think in terms of familiar examples.
+
+Let's consider one from the phyisical world -- access to the various rooms of a 🏠 house -- and compare it to a digital example of a Blog.
+
+#### RBAC Example: House
+
+Consider a 🏠 while you are away on vacation.
+
+You are the **_owner_** and have given out :key: keys to your **neighbor** and a **plumber** that unlock the 🏠 🚪 door.
```
Comment (thedavidprice) [inline cookbook/Role-based_Access_Control.md:36]:
```suggestion
You are the **_owner_** and have given out 🔑 keys to your **neighbor** and a **plumber** that unlock the 🏠 🚪 door.
```

**Review by thedavidprice — APPROVED** [review]

**Comment by thedavidprice** [comment]
This is just exceptional @dthyresson Thank you! 🚀

**Pull request merged**